addons/website/static/src/components/configurator/configurator.js

```diff
@@ -184,25 +184,42 @@ class DescriptionScreen extends Component {
      * @param {function} response callback which takes the data to suggest as argument
      */
     _autocompleteSearch(request, response) {
-        const lcTerm = request.term.toLowerCase();
+        const terms = request.term.toLowerCase().split(" ");
         const limit = 15;
-        const matches = this.state.industries.filter((val) => {
-            return val.label.startsWith(lcTerm);
+        const sortLimit = 7;
+        // `this.state.industries` is already sorted by hit count (from IAP).
+        // That order should be kept after manipulating the recordset.
+        let matches = this.state.industries.filter((val, index) => {
+            // To match, every term should be contained in either the label or a
+            // synonym
+            const candidates = [val.label, ...(val.synonyms || '').split(" ")];
+            for (const term of terms) {
+                if (!candidates.some(el => el.includes(term))) {
+                    return false;
+                }
+            }
+            val.wordCount = val.label.split(" ").length;
+            val.hitCountOrder = index;
```

Comment:
Note that those lines should probably be moved inside the `getInitialState` to alter it after the RPC only once, and not on every autocomplete.